curvefs/src/tools/query/curvefs_inode_query.cpp

```diff
@@ -0,0 +1,185 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2022-03-25
+ * Author: chengyi01
+ */
+
+#include "curvefs/src/tools/query/curvefs_inode_query.h"
+
+#include "src/common/string_util.h"
+
+DECLARE_string(metaserverAddr);
+DECLARE_string(poolId);
+DECLARE_string(copysetId);
+DECLARE_string(partitionId);
+DECLARE_string(fsId);
+DECLARE_string(inodeId);
+
+namespace curvefs {
+namespace tools {
+namespace query {
+
+void InodeQueryTool::PrintHelp() {
+    CurvefsToolRpc::PrintHelp();
+    std::cout << " -poolId=" << FLAGS_poolId
+              << " -copysetId=" << FLAGS_copysetId
+              << " -partitionId=" << FLAGS_partitionId
+              << " -fsId=" << FLAGS_fsId << " -inodeId=" << FLAGS_inodeId
+              << " [-metaserverAddr=" << FLAGS_metaserverAddr << "]";
+    std::cout << std::endl;
+}
+
+void InodeQueryTool::AddUpdateFlags() {
+    AddUpdateFlagsFunc(curvefs::tools::SetMetaserverAddr);
+}
+
+int InodeQueryTool::Init() {
+    if (CurvefsToolRpc::Init() != 0) {
+        return -1;
+    }
+
+    curve::common::SplitString(FLAGS_metaserverAddr, ",", &hostsAddr_);
+
+    std::vector<std::string> poolsId;
+    curve::common::SplitString(FLAGS_poolId, ",", &poolsId);
+
+    std::vector<std::string> copysetsId;
+    curve::common::SplitString(FLAGS_copysetId, ",", &copysetsId);
+
+    std::vector<std::string> partitionId;
+    curve::common::SplitString(FLAGS_partitionId, ",", &partitionId);
+
+    std::vector<std::string> fsId;
+    curve::common::SplitString(FLAGS_fsId, ",", &fsId);
+
+    std::vector<std::string> inodeId;
+    curve::common::SplitString(FLAGS_inodeId, ",", &inodeId);
+
+    if (poolsId.size() != copysetsId.size() ||
+        poolsId.size() != partitionId.size() || poolsId.size() != fsId.size() ||
+        poolsId.size() != inodeId.size()) {
+        std::cout << "poolId, copysetId, partitionId, fsId, inodeId must be "
+                     "the same size"
+                  << std::endl;
+        return -1;
+    }
+
+    for (unsigned i = 0; i < poolsId.size(); ++i) {
+        curvefs::metaserver::GetOrModifyS3ChunkInfoRequest request;
+        request.set_poolid(std::stoi(poolsId[i]));
+        request.set_copysetid(std::stoi(copysetsId[i]));
+        request.set_partitionid(std::stoi(partitionId[i]));
+        request.set_fsid(std::stoi(fsId[i]));
+        request.set_inodeid(std::stoi(inodeId[i]));
+        request.set_returns3chunkinfomap(true);
+        AddRequest(request);
+    }
+
+    service_stub_func_ = std::bind(
+        &curvefs::metaserver::MetaServerService_Stub::GetOrModifyS3ChunkInfo,
```

Comment:
If you want to get the whole s3infomap may be need using stream rpc api when it is too larger. And you can refer to the implementation in metaserver_client.cpp.

Reply:
> If you want to get the whole s3infomap may be need using stream rpc api when it is too larger. And you can refer to the implementation in metaserver_client.cpp.

fix